File: ip-validate.go
package kata

import (
	"strconv"
	"strings"
)

// IsValidIP ...
func IsValidIP(ip string) bool {
	nums := strings.Split(ip, ".")
	if len(nums) != 4 {
		return false
	}
	for _, value := range nums {
		n, err := strconv.Atoi(value)
		if err != nil || n < 0 || n > 255 || len(value) != len(strconv.Itoa(n)) {
			return false
		}
	}

	return true
}

File: ip-validate.js
// const isValidIP = str => {

// 	const octetCheck = /^(25[0-5]|2[0-4]\d|1\d\d|[1-9]\d|\d)$/
// 	const splitStr = str.split('.')
// 	if (splitStr.length !== 4) {
// 		return false
// 	}

// 	for (let i in splitStr) {
// 		if (!octetCheck.test(splitStr[i])) {
// 			return false
// 		}
// 	}

// 	return true
// }

const isValidIP = str => /^(([1-9]?\d|1\d\d|2[0-4]\d|25[0-5])(\.(?!$)|$)){4}$/.test(str);
